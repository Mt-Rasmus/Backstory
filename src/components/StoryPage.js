
import React from 'react';
import StoryModal from './StoryModal';

export default class StoryPage extends React.Component {

   constructor(props) {
      super(props);
      this.state = {
         solved: false,
         showStoryModal: false
      };
   }

   handleShowStoryModal = () => {
      this.setState({ showStoryModal: true });
   }  

   handleCloseStoryModal = () => {
      this.setState({ showStoryModal: false });
   }

   toggleSolved = () => {
      this.setState({ solved: !this.state.solved });
   }

   render() {
      return (
         <div>
            <div onClick={this.handleShowStoryModal} className="content-container">
               <p>{this.props.story.title}</p>
            </div>
            <div>
               <input type="checkbox" id="vehicle1" onClick={this.toggleSolved}></input>
               <label htmlFor="vehicle1">solved</label><br></br>
            </div>
            <div>
               <StoryModal 
                  showModal={this.state.showStoryModal} 
                  handleCloseStoryModal={this.handleCloseStoryModal} 
                  toggleSolved={this.toggleSolved}
                  story={this.props.story}
               />           
            </div>
         </div>
      )
   }
}

