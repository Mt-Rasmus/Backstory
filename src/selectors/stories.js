
import database from '../firebase/firebase';

export const getStories = () => {
   return database.ref(`stories`).once("value").then((snapshot) => {
      return snapshot.val();
   })   
}