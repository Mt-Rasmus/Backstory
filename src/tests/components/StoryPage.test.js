
import React from 'react';
import StoryPage from '../../components/StoryPage';
import { shallow } from 'enzyme';
import stories from '../../fixtures/teststories'; // importing test data

test('Should render StoryPage correctly', () => {
   const wrapper = shallow(<StoryPage story={stories[0]}/>);
   expect(wrapper).toMatchSnapshot();
});
