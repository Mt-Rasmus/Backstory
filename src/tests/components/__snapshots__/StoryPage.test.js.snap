// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render StoryPage correctly 1`] = `
<div>
  <div
    className="content-container"
    onClick={[Function]}
  >
    <p>
      Title of first story
    </p>
  </div>
  <div>
    <input
      id="vehicle1"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="vehicle1"
    >
      solved
    </label>
    <br />
  </div>
  <div>
    <StoryModal
      handleCloseStoryModal={[Function]}
      showModal={false}
      story={
        Object {
          "backstory": "The backstory",
          "id": "1",
          "premise": "A premise to a story",
          "title": "Title of first story",
        }
      }
      toggleSolved={[Function]}
    />
  </div>
</div>
`;
